sonar-slang-plugin/src/test/java/com/sonarsource/slang/kotlin/KotlinConverterTest.java

```diff
@@ -102,6 +104,35 @@ public void testNullParameterNames() {
       .isEquivalentTo(kotlinStatement("for ((a, b) in container) {a}"));
   }
 
+  @Test
+  public void testVariableDeclaration() {
+    Tree varX = kotlinStatement("var x : Int");
+    Tree valY = kotlinStatement("val y : Int");
+    assertTree(varX).isInstanceOf(VariableDeclarationTree.class);
+    assertTree(valY).isInstanceOf(VariableDeclarationTree.class);
+    assertTree(((VariableDeclarationTree) varX).identifier()).isIdentifier("x");
+    assertThat(((VariableDeclarationTree) varX).isVal()).isFalse();
+    assertTree(((VariableDeclarationTree) valY).identifier()).isIdentifier("y");
+    assertThat(((VariableDeclarationTree) valY).isVal()).isTrue();
+    assertTree(varX).isEquivalentTo(kotlinStatement("var x: Int"));
+    assertTree(varX).isNotEquivalentTo(kotlinStatement("var y: Int"));
+    assertTree(varX).isNotEquivalentTo(kotlinStatement("val x: Int"));
+    assertTree(varX).isNotEquivalentTo(kotlinStatement("var x: Boolean"));
+  }
+
+  @Test
+  public void testVariableDeclarationWithInitializer() {
+    Tree varX = kotlinStatement("var x : Int = 0");
+    Tree valY = kotlinStatement("val x : Int = \"4\"");
+    assertTree(varX).isInstanceOf(VariableDeclarationTree.class);
+    assertTree(valY).isInstanceOf(VariableDeclarationTree.class);
+    assertThat(((VariableDeclarationTree) varX).initializer()).isInstanceOf(LiteralTree.class);
+    assertThat(((VariableDeclarationTree) valY).initializer()).isInstanceOf(StringLiteralTree.class);
+    assertTree(varX).isNotEquivalentTo(valY);
```

Comment:
One `isEquivalentTo` assert would be nice too here, to ensure equivalence works correctly for declarations with initializers